slack-connector/internal/registration/payloadbuilder.go

```diff
@@ -0,0 +1,74 @@
+package registration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/kyma-incubator/hack-showcase/slack-connector/internal/apperrors"
+)
+
+const (
+	specificationURL          = "https://raw.githubusercontent.com/kyma-incubator/hack-showcase/slack-connector-boilerplate/slack-connector/internal/registration/configs/slackopenapi.json" // to edit after merge
+	applicationRegistryPrefix = "http://application-registry-external-api.kyma-integration.svc.cluster.local:8081/"
+	applicationRegistrySuffix = "-app/v1/metadata/services"
+	applicationRegistryFormat = "%s%s%s"
+)
+
+//FileReader is an interface used to allow mocking file reading
+type FileReader interface {
+	Read(string) ([]byte, error)
+}
+
+//NewPayloadBuilder creates a serviceDetailsPayloadBuilder instance
+type payloadBuilder struct {
+	builder         PayloadBuilder
+	fileReader      FileReader
+	applicationName string
+}
+
+//NewPayloadBuilder creates a serviceDetailsPayloadBuilder instance
+func NewPayloadBuilder(fr FileReader, appName string) payloadBuilder {
+	return payloadBuilder{fileReader: fr, applicationName: appName}
+}
+
+//Build creates a ServiceDetails structure with provided API specification URL
+func (r payloadBuilder) Build() (ServiceDetails, error) {
+
+	var jsonBody = ServiceDetails{
+		Provider:    "Kyma",
+		Name:        r.applicationName,
+		Description: "Slack Connector, which is used for registering Slack API in Kyma",
+		API: &API{
+			TargetURL:         "https://slack.com/api",
+			RequestParameters: &RequestParameters{Headers: &Headers{CustomHeader: []string{os.Getenv("SLACK_BOT_TOKEN")}}},
```

Comment:
Getting env value here blocks you from mocking it in tests. In your test you are checking all json values but not RequestParameters. Adding a field to payloadBuilder struct and getting env value in main would resolve that problem.

Reply:
If you don't want to do that in main you can create a GetToken function, and then mock it in your tests. That's how it's done in the github connector. You can check it right [here](https://github.com/kyma-incubator/hack-showcase/blob/master/github-connector/internal/handlers/webhookhandler.go#L43)